src/index.js

```diff
@@ -319,6 +332,11 @@ class Libp2p extends EventEmitter {
    * @private
    */
   _onDidStart () {
+    // Peer discovery
+    if (this._modules.peerDiscovery) {
+      this._setupPeerDiscovery()
+    }
```

Comment:
Is this if necessary? I think it would be good to have any necessary checks inside of `_setupPeerDiscovery`. I think it will be much easier for us to avoid bugs if we have the functions handle their checks.